src/java/za/co/mip/ablduck/utilities/HTMLGenerator.java

```diff
@@ -0,0 +1,376 @@
+/**
+ * Copyright 2017 MIP Holdings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package za.co.mip.ablduck.utilities;
+
+import java.util.List;
+import java.util.HashMap;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import org.apache.tools.ant.BuildException;
+
+import com.google.gson.annotations.SerializedName;
+
+import za.co.mip.ablduck.models.SourceJSObject;
+import za.co.mip.ablduck.models.generic.DeprecatedObject;
+import za.co.mip.ablduck.models.generic.MetaObject;
+import za.co.mip.ablduck.models.source.MemberObject;
+import za.co.mip.ablduck.models.source.ParameterObject;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+
+public class HTMLGenerator {
+    private static Pattern returnType = Pattern.compile("\\):(.*)");
+    private HashMap<String, SourceJSObject> classes;
+            
+    public String getClassHtml (HashMap<String, SourceJSObject> allclasses, SourceJSObject cls) {

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The type of the "allclasses" object should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1319)